pintc/tests/optimizations/dead_select_elimination.pnt

```diff
@@ -0,0 +1,92 @@
+predicate test {
+    // select exprs that should be optimized out
+    var a = true ? 1 : 2;
+    var b = false ? 1 : 2;
+    var c = (1 + 2 == 3) ? 1 : 2;
+    var d = (1 < 2) ? 1 : 2;
+
+    // should this be removed? The evaluator can't handle this right now
+    var e = 2 < 1;
+    var f = e ? 1 : 2;
```

Comment:
Comment covers it here. We faced a similar issue with #834 where we currently don't remove `constraints` that have `vars` in the keys for the `constraint`.

It's currently just a limitation of the `evaluator` so maybe that should be opened as a new issue? Or maybe we don't want to remove things like this and `evaluator` is working as intended?

Reply:
Yeah that's a more "advanced" evaluation.. let's not worry about it just yet, though it wouldn't be too difficult to do.